contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    enum TransferAction {INTERVAULT, WITHDRAW}
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        address depositContract;
+        uint32 timestamp;
+        uint256 amount;
+        TransferAction action;
+    }
+
+    using SafeERC20 for IERC20;
+
+    mapping(address => Transfer[]) public qTransfer;
+    mapping(address => uint256) public totalAmount;
+
+    address public wethVault;
+    address public stethVault;
+
+    event Disburse(address vault, Transfer txn, uint256 portion);
+
+    function initialize(address _wethVault, address _stethVault)
+        external
+        initializer
+    {
+        __ReentrancyGuard_init();
+        __Ownable_init();
+
+        transferOwnership(msg.sender);
+
+        wethVault = _wethVault; // Can be the native vault (weth or wavax)
+        stethVault = _stethVault; // Must ONLY be steth vault.  (savax vault returns erc20, set as keeper)
+    }
+
+    function getInterVaultBalance(address vault)
+        private
+        view
+        returns (uint256 balance)
+    {
+        // Case 1. stETH - Withdrawing gives stETH
+        // Case 2. WETH - Withdrawing gives ETH
+        // Case 3. erc20 token
+        if (vault == stethVault) {
+            address steth = IRibbonVault(vault).STETH();
+            balance = IERC20(steth).balanceOf(address(this));
+        } else if (vault == wethVault) {
+            balance = address(this).balance;
+        } else {
+            Vault.VaultParams memory vaultParams =
+                IRibbonVault(vault).vaultParams();
+            balance = IERC20(vaultParams.asset).balanceOf(address(this));
+        }
+    }
+
+    function withdrawToCreditor(
+        address vault,
+        address creditor,
+        uint256 portion
+    ) private {
+        if (vault == stethVault) {
+            address steth = IRibbonVault(vault).STETH();
+            IERC20(steth).transfer(creditor, portion);
+        } else if (vault == wethVault) {
+            payable(creditor).transfer(portion);
```

Comment:
The better way to send ether is to use `payable(creditor).call{value: portion}("")`